lib/vector-core/src/test_util.rs

```diff
@@ -0,0 +1,11 @@
+use std::fs::File;
+use std::path::Path;
+
+pub fn open_fixture(path: impl AsRef<Path>) -> crate::Result<serde_json::Value> {
+    let test_file = match File::open(path) {
+        Ok(file) => file,
+        Err(e) => return Err(e.into()),
+    };
+    let value: serde_json::Value = serde_json::from_reader(test_file)?;
+    Ok(value)
```

Comment:
Which makes this a one-liner:
```suggestion
    serde_json::from_reader(File::open(path)?).map_err(Into::into)
```